[BUG] Can't compile outside of source tree

### GZDoom version

1a752f93a96539dbd3a9c947a15fc5c37598c6eb and later

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Linux x86_64

### Please describe your specific OS version

Arch

### Relevant hardware info

N/A

### Have you checked that no other similar issue already exists?

- [x] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

I get the following error upon trying to compile in a directory outside of the source tree:
```
FAILED: src/CMakeFiles/revision_check /home/marrub/bin/other/uzdoom/src/CMakeFil
es/revision_check
cd /home/marrub/bin/other/uzdoom && /usr/bin/cmake -P /home/marrub/src/games/UZD
oom/tools/updaterevision/UpdateRevision.cmake /home/marrub/bin/other/uzdoom/giti
nfo.h
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Not a git repo! Set version tag by setting GIT_DESCRIBE env var
CMake Error at /home/marrub/src/games/UZDoom/tools/updaterevision/UpdateRevision
.cmake:100 (message):
  Failed to get commit info:
Call Stack (most recent call first):
  /home/marrub/src/games/UZDoom/tools/updaterevision/UpdateRevision.cmake:109 (m
ain)
```

### Steps to reproduce the behaviour.

1. Create a directory outside of the source tree, so for instance with `~/src/games/UZDoom` use `~/bin/other/uzdoom`.
2. Run CMake in the directory, which will succeed just fine: `cmake ~/src/games/UZDoom -DCMAKE_BUILD_TYPE=RelWithDebInfo -G Ninja`
3. Build (`ninja`), which should error.

### Your configuration

```ini

```

### Provide a Log

_No response_

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -318,7 +318,7 @@ option(BUILD_FLATPAK "Enable Flatpak-specific build options" OFF)
 
 add_custom_target( revision_check ALL
 	COMMAND "${CMAKE_COMMAND}" -P "${CMAKE_SOURCE_DIR}/tools/updaterevision/UpdateRevision.cmake" "${CMAKE_BINARY_DIR}/gitinfo.h"
-	WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
+	WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
 )
 
 # required libraries

--- a/tools/updaterevision/UpdateRevision.cmake
+++ b/tools/updaterevision/UpdateRevision.cmake
@@ -18,10 +18,13 @@ function(query_repo_info)
 	# are we git?
 	execute_process(
 		COMMAND git rev-parse --is-inside-work-tree
-		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 		RESULT_VARIABLE is_git
 		OUTPUT_QUIET
 	)
+
+	set(Tag "unknown")
+	string(TIMESTAMP Timestamp "%Y-%m-%d %H:%M:%S %z")
+	set(Hash "0000000")
 
 	if(DEFINED ENV{GIT_DESCRIBE})
 		# from env
@@ -35,49 +38,41 @@ function(query_repo_info)
 		string(REGEX MATCH "-g([0-9a-fA-F]+)" match_result "${Tag}")
 		if(match_result)
 			set(Hash "${CMAKE_MATCH_1}")
-		else()
-			set(Hash "0000000")
 		endif()
-
-		string(TIMESTAMP Timestamp "%Y-%m-%d %H:%M:%S %z")
 	elseif(is_git EQUAL "0")
 		# from git
 		execute_process(
 			COMMAND git describe --tags --dirty=-m
 			RESULT_VARIABLE Error
-			OUTPUT_VARIABLE Tag
+			OUTPUT_VARIABLE Temp
 			ERROR_QUIET
 			OUTPUT_STRIP_TRAILING_WHITESPACE
 		)
 
 		if(NOT "${Error}" STREQUAL "0")
-			message(STATUS "No git tags found! Set version tag by setting GIT_DESCRIBE env var")
-			ret_var(Error)
-			return()
+			message(STATUS "No git tags found! Using fallback '${Tag}'")
+		else()
+			set(Tag "${Temp}")
 		endif()
 
 		execute_process(
 			COMMAND git log -1 "--format=%ai;%H"
 			RESULT_VARIABLE Error
-			OUTPUT_VARIABLE CommitInfo
+			OUTPUT_VARIABLE Temp
 			ERROR_QUIET
 			OUTPUT_STRIP_TRAILING_WHITESPACE
 		)
 
 		if(NOT "${Error}" STREQUAL "0")
-			message(STATUS "No git commits found! Set version tag by setting GIT_DESCRIBE env var")
-			ret_var(Error)
-			return()
+			message(STATUS "No git commits found! Using fallback '${Hash}'")
+		else()
+			string(REPLACE ";" ";" CommitInfo "${Temp}")
+			list(GET CommitInfo 0 Timestamp)
+			list(GET CommitInfo 1 Hash)
 		endif()
 
-		string(REPLACE ";" ";" CommitInfoList "${CommitInfo}")
-		list(GET CommitInfoList 0 Timestamp)
-		list(GET CommitInfoList 1 Hash)
 	else()
-		# helpful message
 		message(STATUS "Not a git repo! Set version tag by setting GIT_DESCRIBE env var")
-		ret_var(Error)
-		return()
 	endif()
 
 	ret_var(Tag)

